peepingtom/datablocks/multiblocks/_tests/test_meshblock.py

```diff
@@ -21,20 +21,20 @@
 
 def test_meshblock_instantiation():
     # check that meshblock can be instantiated properly
-    block = MeshBlock(vertices, faces)
+    block = MeshBlock(vertices_data=vertices, faces_data=faces)
     assert isinstance(block, MeshBlock)
     assert hasattr(block, 'vertices')
     assert hasattr(block, 'faces')
 
 
 def test_triangles():
-    block = MeshBlock(vertices, faces)
+    block = MeshBlock(vertices_data=vertices, faces_data=faces)
     triangles = block.triangles
-    assert triangles.shape[0] == faces.shape[0]
-    assert triangles.shape[2] == 3
+    assert len(triangles) == len(faces)
+    assert triangles.shape[-1] == 3
 
 
 def test_midpoints():
-    block = MeshBlock(vertices, faces)
+    block = MeshBlock(vertices_data=vertices, faces_data=faces)
```

Comment:
the more I see this the less I love the `*_data` kwargs, was there a reason for not using `points` or `faces`?

More broadly, does this preclude the possibility of multiple of the same blocktype within one multiblock?

Reply:
Right now, if a multiblock implements:
```python
_block_types = {'steve': SteveBlock}
```
and `SteveBlock` has a `data` and a `stuff` kwarg, the signature will be `steve_data, steve_stuff` . You *could* get around this by special-casing kwargs called `data` to simply fall back to `steve`.

This does not preclude multiblock with blocks of the same type, but does preclude blocks with the same name (which is intended). Indeed, [`DipoleBlock` does it](https://github.com/gutsche-lab/peepingtom/blob/0b96b564ea6ecf22729b46297a8300facb8fbdfa/peepingtom/datablocks/multiblocks/dipoleblock.py#L13).